lib/logstash/plugin_mixins/jdbc/jdbc.rb

```diff
@@ -55,6 +55,9 @@ def setup_jdbc_config
       # Be aware that ordering is not guaranteed between queries.
       config :jdbc_paging_enabled, :validate => :boolean, :default => false
 
+      # Whether to use manual mode during the JDBC paging
+      config :jdbc_paging_manual_mode, :validate => :boolean, :default => false
```

Comment:
sorry for the late review, did you guys consider having better naming for the feature? e.g.

`jdbc_paging_enabled => true jdbc_paging_mode => offset` any chance this reads better (than `"manual"`)?
(`jdbc_paging_mode => count` would de the default value)

should also keep things open for changing the paging strategy again (offset paging is a bit of an anti pattern although for mostly static data and LS' needs of at least once delivery it's sufficient), wdyt?

Reply:
@kares I like your suggestion,  instead of using a boolean flag defined a string value for the strategy is more readable.
However this feature doesn't cover only static data but also use cases where the `Sequel` automatic paging doesn't apply correctly, mostly:
- procedure calls with paging limits
- paging in inner queries in case of nested SQL queries

Reply:
I see, maybe we need better naming there - haven't thought about it that much.
however the manual just read a bit weird - had no idea what that could mean SQL wise.

what I did is check Sequel's [`each_page`](https://github.com/evanfarrar/opensprints/blob/master/lib/sequel/lib/sequel/extensions/pagination.rb#L22-L24) and it's using a `COUNT(...)` initial query (the details on how the statement is transformed into a count is usually adapter specific), haven't checked any adapter specific overrides for `each_page`.

Reply:
@kares to me the naming you are proposing is good, better than the boolean flag. So I would use it.

Reply:
@kares @andsel @karenzone Thanks for your time. I also think this suggestion is good and will improve according to it.
I have a little question about the name.
Actually, this feature involves two options:
1. Not to add paging condition automatically, but explicitly customized by user.
1. Not to use count query first

The first one is the main purpose of this feature.
So, is it easy for users to understand the two values of 'count' and 'offset'?
If everyone don't have other suggested names, I will implement it according to current suggestion.

Reply:
Good point, we could use 
- `automatic` for `dbc_paging_mode => count` and 
- `explicit` for `jdbc_paging_mode => offset`

@kares is it a better naming in your opinion?

Reply:
those sound better instead of automatic maybe `jdbc_paging_mode => auto` to choose the best available strategy based on the driver (Sequel does this). `jdbc_paging_mode => explicit` or `manual` both sounds okay to me.